### ISSUE TITLE
IoT Topic Rule fails with E1029

### ISSUE THREAD

**Issue by carlosaln** [aws-cloudformation/cfn-python-lint#1064]
*cfn-lint version: 0.22.4*

*Description of issue.*

When using AWS IoT substitution templates (in my case, for IoT SQL functions) within Cloud Formation, it is necessary to use the dollar sign and curly braces (For example `${topic()}`). This gets misinterpreted as a Fn::Sub Parameter which throws an E1029 error.

Please provide as much information as possible:
* Template linting issues:
  * Please provide a CloudFormation sample that generated the issue.
```yaml
  IotTopicRule: 
    Type: AWS::IoT::TopicRule
    Properties: 
      RuleName: IotTopicRule
      TopicRulePayload:
        RuleDisabled: false
        Sql: !Sub "SELECT * FROM 'some-topic'"
        Actions: 
          - 
            Kinesis: 
              RoleArn: !Sub '${topicRole.Arn}'
              StreamName: !Ref MyKinesisStream
              PartitionKey: "${topic()}" # error happens here
```
  * If present, please add links to the (official) documentation for clarification.

AWS IoT substitution templates are explained here: https://docs.aws.amazon.com/iot/latest/developerguide/iot-substitution-templates.html

How !Sub uses variables (which `cfn-lint` looks for) is found here: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-sub.html#w2ab1c21c24c59b7

  * Validate if the issue still exists with the latest version of `cfn-lint` and/or the latest Spec files

Yes

**Comment by kddejong** [comment]
Released with v0.23.0
